build_release_image.sh

```diff
@@ -0,0 +1,119 @@
+help() {
+    echo "Build an Openshift Release Image"
+    echo "Usage:                    build_release_image.sh [options] <Images> <quay.io username>"
+    echo ""
+    echo "Options:"
+    echo "-h, --help                show this message"
+    echo "-t, --tag <tag>           push to a custom tag in your origin release image repo, default: mco"
+    echo ""
+    echo "Images: you must include at least one"
+    echo "-m, --mco                 build and push dependancies for mco to quay.io/<username>/image then"
+    echo "                              build a release image from that repo."
+    echo "-c, --capo [tag]          build a capo image locally, push it to quay.io/<username>/capo:[tag] then"
+    echo "                              build a release image from that repo."
+}
+
+TAG="mco"
+USERNAME=""
+CAPO=""
+MCO=""
+set -x
+# Parse Options
+while :; do
+    case $1 in
+        -h| --help )
+            help
+            exit 0
+            ;;
+        -t | --tag )
+            if [ -z "$2" ]; then
+                echo "Error: tag option passed without a tag value. Please use -h, or --help to see usage"
+                exit 1
+            fi
+            TAG=$2
+            ;;
+        -c | --capo )
+            if [ -z "$2" ]; then
+                CAPO="latest"
+            elif [[ $2 =~ ^\-.*|^\-\-.* ]];then
+                CAPO="latest"
+            else
+                CAPO="$2"
+            fi
+            ;;
+        -m | --mco )
+            MCO="true"
+            ;;
+        \? )
+            printf 'WARN: Unknown option : %s\n' "$1" >&2
+            exit 1
+            ;;
+        * )
+            shift
+            break
+            ;;
+    esac
+
+    shift
+done
+
+if [ -z "$CAPO" ] && [ -z "$MCO" ];then
+    echo "No images provided, can't build release image. Please use -h, or --help to see usage"
```

Comment:
Why? it's perfectly valid to build a release image without our custom payload, we shouldn't prevent it. This way we don't have to update the above check each time we add support for building other images ;-)

Reply:
It actually fails if you dont :), at least in its current form

Reply:
then let the `oc adm` command fail. This will be immediate since we're not building images.
It's better to expose the real issue from `oc adm` (that is hopefully self explanatory) than make our script more complex to hack around other tools limitations.

Reply:
I think since I defined the UI to require at least one image be specified, it is fine to check for this case of incorrect usage. The error from oc adm is ```error: must specify image mappings``` which isnt terrible, but its not abundantly clear what they did wrong. I'd much rather spell it out for people so nobody is scratching their head.

Reply:
Alright then, I agree the error that oc adm throws is not the most obvious...

Reply:
fixed it

Reply:
It can now take no arguments